test/peer-store/persisted-peer-store.spec.js

```diff
@@ -0,0 +1,374 @@
+
+      const queryParams = {
+        prefix: '/peers/'
+      }
+      for await (const _ of datastore.query(queryParams)) { // eslint-disable-line
+        throw new Error('Datastore should be empty')
+      }
+
+      // Add data for second book
+      peerStore.addressBook.set(peers[1], multiaddrs)
+
+      expect(spyDirty).to.have.property('callCount', 4)
+      expect(spyDs).to.have.property('callCount', 1)
+      expect(peerStore._dirtyPeers.size).to.equal(0) // Reset
+
+      // Should have two peer records stored in the datastore
+      let count = 0
+      for await (const _ of datastore.query(queryParams)) { // eslint-disable-line
+        count++
+      }
+      expect(count).to.equal(2)
+      expect(peerStore.peers.size).to.equal(2)
+    })
+
+    it('should commit on stop if threshold was not reached', async () => {
+      const spyDirty = sinon.spy(peerStore, '_addDirtyPeer')
+      const spyDs = sinon.spy(datastore, 'batch')
+
+      const protocols = ['/ping/1.0.0']
+      const [peer] = await peerUtils.createPeerId()
+
+      await peerStore.start()
+
+      // Add Peer data in a booka
+      peerStore.protoBook.set(peer, protocols)
+
+      expect(spyDs).to.have.property('callCount', 0)
+      expect(spyDirty).to.have.property('callCount', 1)
+      expect(peerStore._dirtyPeers.size).to.equal(1)
+
+      const queryParams = {
+        prefix: '/peers/'
+      }
+      for await (const _ of datastore.query(queryParams)) { // eslint-disable-line
+        throw new Error('Datastore should be empty')
+      }
+
+      await peerStore.stop()
+
+      expect(spyDirty).to.have.property('callCount', 1)
+      expect(spyDs).to.have.property('callCount', 1)
+      expect(peerStore._dirtyPeers.size).to.equal(0) // Reset
+
+      // Should have one peer record stored in the datastore
+      let count = 0
+      for await (const _ of datastore.query(queryParams)) { // eslint-disable-line
+        count++
+      }
+      expect(count).to.equal(1)
+      expect(peerStore.peers.size).to.equal(1)
+    })
+  })
+})
+
+describe('libp2p.peerStore (Persisted)', () => {
+  describe('disabled by default', () => {
+    let libp2p
+
+    before(async () => {
+      [libp2p] = await peerUtils.createPeer({
+        started: false
+      })
+    })
+
+    afterEach(() => libp2p.stop())
+
+    it('should not have have persistence capabilities', async () => {
+      await libp2p.start()
+      expect(libp2p.peerStore._dirtyPeers).to.not.exist()
+      expect(libp2p.peerStore.threshold).to.not.exist()
+    })
+  })
+
+  describe('enabled', () => {
+    let libp2p
+    let memoryDatastore
+
+    beforeEach(async () => {
+      memoryDatastore = new MemoryDatastore()
+      ;[libp2p] = await peerUtils.createPeer({
+        started: false,
+        config: {
+          datastore: memoryDatastore,
+          peerStore: {
+            persistence: true,
+            threshold: 2 // trigger on  second peer changed
```

Comment:
Just something to think about, we should probably do/recommend a threshold of 1 in browsers. There is no interval and browser nodes won't "stop" properly out of the box, so we could end up with unflushed records when the window is closed

Reply:
We should probably discuss the addition of an interval for persist it. However, this will also not be a bullet proof solution. I can create an issue for this as an action item.

Reply:
Meanwhile, I added a note for that in the CONFIGURATION.md